src/Controller/AppController.php

```diff
@@ -89,6 +85,32 @@ public function beforeFilter(Event $event): ?Response
         return null;
     }
 
+    /**
+     * Return route array for login redirect.
+     * When request is not a get, return route without redirect.
+     * When request uri path has attribute webroot, return route without redirect.
+     * Return route with redirect, otherwise.
+     *
+     * @return array
+     */
+    protected function loginRedirectRoute(): array
+    {
+        $route = ['_name' => 'login'];
+
+        // if request is not a get, return route without redirect.
+        if (!$this->request->is('get')) {
+            return $route;
+        }
+
+        // if redirect is webroot, return route without redirect.
+        $redirect = $this->request->getUri()->getPath();
```

Comment:
Usually requests lie `JS/CSS` or other resources in webroot should not be passed to the PHP app, we should ignore this case.
Request URL could be wrong, but we can do nothing about that...

Reply:
> Usually requests lie `JS/CSS` or other resources in webroot should not be passed to the PHP app, we should ignore this case.
> Request URL could be wrong, but we can do nothing about that...

A misunderstanding: this `webroot` is not js/css folder; it's the app webroot string.
Maybe phpdoc confused. I updated it.

Reply:
> Usually requests lie `JS/CSS` or other resources in webroot should not be passed to the PHP app, we should ignore this case.
> Request URL could be wrong, but we can do nothing about that...

A misunderstanding: this `webroot` is not js/css folder; it's the app webroot string.
Maybe phpdoc confused. I updated it.